internal/component/statefulset/builder.go

```diff
@@ -137,12 +137,12 @@ func (b *stsBuilder) getStatefulSetLabels() map[string]string {
 
 func (b *stsBuilder) createStatefulSetSpec(ctx component.OperatorContext) error {
 	err := b.createPodTemplateSpec(ctx)
-	b.sts.Spec.Replicas = ptr.To(utils.IfConditionOr(druidv1alpha1.IsEtcdRuntimeComponentCreationEnabled(b.etcd.ObjectMeta), b.replicas, 0))
```

Comment:
Not sure if this was intended to be part of this PR. Please feel free to raise a new PR with code changes if you would like to cleanup some of the old code, so that we can keep the scope of this PR limited.

Reply:
It is intended, I will not open a PR for every few changed lines.